AWSAppSyncClient/AWSAppSyncClient.swift

```diff
@@ -631,8 +647,65 @@ public protocol AWSAppSyncOfflineMutationDelegate {
     func mutationCallback(recordIdentifier: String, operationString: String, snapshot: Snapshot?, error: Error?) -> Void
 }
 
+public struct AppSyncConnectionInfo {
+    public let isConnectionAvailable: Bool
+    public let isInitialConnection: Bool
+}
+
+internal extension Notification.Name {
+    internal static let appSyncReachabilityChanged = Notification.Name("AppSyncNetworkAvailabilityChangedNotification")
+}
+
+class AWSAppSyncNetworkStatusChangeNotifier {
+    var reachability: Reachability?
+    var allowsCellularAccess: Bool = true
+    var isInitialConnection: Bool = true
+    
+    static func setupSharedInstance(host: String, allowsCellular: Bool) {
+        sharedInstance = AWSAppSyncNetworkStatusChangeNotifier(host: host, allowsCellular: allowsCellular)
+    }
+    
+    static var sharedInstance: AWSAppSyncNetworkStatusChangeNotifier?
+    
+    init(host: String, allowsCellular: Bool) {
```

Comment:
If this is a singleton, should the `init` be private?

Reply:
Yup, fixing it.